mobilecoind/src/service.rs

```diff
@@ -355,16 +355,17 @@ impl<T: BlockchainConnection + UserTxConnection + 'static> ServiceApi<T> {
         &mut self,
         request: mc_mobilecoind_api::ReadRequestCodeRequest,
     ) -> Result<mc_mobilecoind_api::ReadRequestCodeResponse, RpcStatus> {
-        let request_payload = RequestPayload::decode(request.get_b58_code())
-            .map_err(|err| rpc_internal_error("RequestPayload.decode", err, &self.logger))?;
+        let request_wrapper = mc_mobilecoind_api::printable::PrintableWrapper::b58_decode(
+            request.get_b58_code().to_string(),
+        )
+        .map_err(|err| rpc_internal_error("PrintableWrapper_b58_decode", err, &self.logger))?;
 
+        let payment_request = request_wrapper.get_payment_request();
```

Comment:
This should check if it has a payment request. I believe there is a method called `request_wrapper.has_payment_request()` that would return False if not, and if that happens we should return an error response.